src/hotspot/share/jvmci/jvmciRuntime.cpp

```diff
@@ -117,34 +117,28 @@ class RetryableAllocationMark {
   }
 };
 
-JRT_BLOCK_ENTRY(void, JVMCIRuntime::new_instance_common(JavaThread* current, Klass* klass, bool null_on_fail))
+JRT_BLOCK_ENTRY(void, JVMCIRuntime::new_instance_or_null(JavaThread* current, Klass* klass))
   JRT_BLOCK;
   assert(klass->is_klass(), "not a class");
   Handle holder(current, klass->klass_holder()); // keep the klass alive
   InstanceKlass* h = InstanceKlass::cast(klass);
   {
-    RetryableAllocationMark ram(current, null_on_fail);
+    RetryableAllocationMark ram(current);
     h->check_valid_for_instantiation(true, CHECK);
-    oop obj;
-    if (null_on_fail) {
-      if (!h->is_initialized()) {
-        // Cannot re-execute class initialization without side effects
-        // so return without attempting the initialization
-        return;
-      }
-    } else {
-      // make sure klass is initialized
-      h->initialize(CHECK);
+    if (!h->is_initialized()) {
+      // Cannot re-execute class initialization without side effects
+      // so return without attempting the initialization
+      return;
```

Comment:
Do we need to call `current->set_vm_result(nullptr)` on these bailout paths?

Reply:
That's done in `~RetryableAllocationMark`.

Reply:
Only for the HAS_PENDING_EXCEPTION case.  What about the !h->is_initialized() case?

Reply:
Good observation - seems like this is an outstanding bug. Can you please address that Yudi.
In practice, I wonder how much this matters as Graal always [clears the object result](https://github.com/oracle/graal/blob/0b61d20b08b1af76bd35cfb673c7be8d33855f51/compiler/src/jdk.graal.compiler/src/jdk/graal/compiler/hotspot/stubs/ForeignCallSnippets.java#L127) after reading it.

Reply:
Good point